Checker/main.fs

```diff
@@ -42,6 +42,21 @@ let canBeCompiled content =
             printfn "compilation failed: %s" info
             false
 
+let testFloatToS () =
+    let input = "1. 10. 100. 1000. 10000. 1000000. 1e+123 0.1 0.01 0.001 0.0001 0.000001 1e-123 -1. -10. -100. -1000. -10000. -1000000. -1e+123 -0.1 -0.01 -0.001 -0.0001 -0.000001 -1e-123".Split(" ")
+    let expected = "1. 10. 100. 1000. 10000. 1e6 1e123 .1 .01 .001 .0001 1e-6 1e-123 -1. -10. -100. -1000. -10000. -1e6 -1e123 -.1 -.01 -.001 -.0001 -1e-6 -1e-123"
+    //let expected = "1. 10. 1e2 1e3 1e4 1e6 1e123 .1 .01 1e-3 1e-4 1e-6 1e-123 -1. -10. -1e2 -1e3 -1e4 -1e6 -1e123 -.1 -.01 -1e-3 -1e-4 -1e-6 -1e-123"
+    // TODO: Make a better floatToS.
+    let actual = input |> Array.map float
+                 |> Array.map Printer.floatToS
+    let a = String.Join(" ", actual)
+    if a = expected then
+        printfn "%s" "Success: floatToS"
+        0
+    else
+        printfn "Failure.\nExpected: %s\nActual:   %s" expected a
+        1
+
```

Comment:
Maybe the golden tests should be enough?

Reply:
See unit/float.frag

Reply:
This code isn't much better than before, it still relies on implementation details of F#'s "%g".
We could do a much better job (e.g. turn 1000 into 1e3).
Many test cases are needed for this kind of code, even more than we have now.
We could use the existing golden tests for this, but Printer.floatToS being independent of the rest of the code makes unit testing feel appropriate.

Reply:
It's unclear whether 1e3 is better than 1000.

Golden tests make it easy to update the results when we change the implementation, and it's consistent all the other tests we have.

Reply:
Here's a more compelling example: currently "100000." isn't minified to "1e5", that's 7 bytes instead of 3.
"1e5" is better than "100000." by at least one measure: uncompressed size.
I understand your point that to many people, compressed size is more interesting, but it's also more difficult to measure reliably.
At some point we might want to introduce flags to select a minimization profile.
Until we do, doesn't it make sense to have the default behavior minimize the uncompressed byte count?